src/main/java/cinema/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,48 @@
+package cinema.controller;
+
+import cinema.model.ShoppingCart;
+import cinema.model.dto.MovieSessionDtoMapper;
+import cinema.model.dto.MovieSessionRequestDto;
+import cinema.model.dto.ShoppingCartDtoMapper;
+import cinema.model.dto.ShoppingCartResponseDto;
+import cinema.service.ShoppingCartService;
+import cinema.service.UserService;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Controller
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final MovieSessionDtoMapper movieSessionDtoMapper;
+    private final ShoppingCartDtoMapper shoppingCartDtoMapper;
+
+    public ShoppingCartController(ShoppingCartService shoppingCartService,
+                                  UserService userService,
+                                  MovieSessionDtoMapper movieSessionDtoMapper,
+                                  ShoppingCartDtoMapper shoppingCartDtoMapper) {
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.movieSessionDtoMapper = movieSessionDtoMapper;
+        this.shoppingCartDtoMapper = shoppingCartDtoMapper;
+    }
+
+    @PostMapping("/movie-sessions")
+    public void addMovieSession(@RequestParam String userEmail,
+                                @RequestBody MovieSessionRequestDto movieSession) {
```

Comment:
I wasn't able to get it works with moviesSession - are you getting MovieSessionId in your RequestDto?